kk/models/base.py

```diff
@@ -81,10 +81,10 @@ class WithCommentsMixin(models.Model):
     """
     Mixin for models which can be commented.
     """
-    n_comments = models.IntegerField(verbose_name=_('Number of comments'), blank=True, default=0)
+    n_comments = models.IntegerField(verbose_name=_('Number of comments'), blank=True, default=0, editable=False)
 
     def recache_n_comments(self):
-        new_n_comments = self.comments.count()
+        new_n_comments = self.comments.exclude(deleted=True).count()
```

Comment:
We'll have to do a pass to do that for all API viewsets anyway. I don't think any regular API should return deleted objects.